### PR TITLE
RT4206: Add CHACHA20 alias for ciphers.

### PR DESCRIPTION
**Pull request by tmshort**
Update ciphers documentation as well (based on -04 rev of ID).

### CODE REVIEW

**Comment by kroeckx** [comment]
I was wondering what happened with this yesterday, since I actually wanted to use it.
+1

File: doc/apps/ciphers.pod [hunk 1]
```diff
@@ -657,6 +661,16 @@ Note: these ciphers can also be used in SSL v3.
  DHE_PSK_WITH_AES_128_CCM_8                DHE-PSK-AES128-CCM8
  DHE_PSK_WITH_AES_256_CCM_8                DHE-PSK-AES256-CCM8
 
+=head2 ChaCha20-Poly1305 cipher suites from draft-ietf-tls-chacha20-poly1305-04, extending TLS v1.2
```
Comment (richsalz) [inline doc/apps/ciphers.pod]: don't we have the rfc yet? :)

**Comment by richsalz** [comment]
You going to merge, or should I?

File: doc/apps/ciphers.pod (hunk 1 above)
Comment (tmshort) [inline doc/apps/ciphers.pod]: It's as of -04, it was different in -03; it could very well change again in -05.

Comment (richsalz) [inline doc/apps/ciphers.pod]: It won't, it's what we're all testing interop with :)  Shrug, whatever.
Also, -05 is out now.

**Comment by kroeckx** [comment]
You seem to like to do that, even when I assign it to myself.  So go ahead.

File: doc/apps/ciphers.pod (hunk 1 above)
Comment (ghedo) [inline doc/apps/ciphers.pod]: FWIW, if you remove the numbers completely, the ietf website redirects to the latest version https://tools.ietf.org/html/draft-ietf-tls-chacha20-poly1305

So the reference won't get out-of-date so quickly.

File: ssl/ssl_locl.h [hunk 2]
```diff
@@ -364,6 +364,7 @@
 
 # define SSL_AES                 (SSL_AES128|SSL_AES256|SSL_AES128GCM|SSL_AES256GCM|SSL_AES128CCM|SSL_AES256CCM|SSL_AES128CCM8|SSL_AES256CCM8)
 # define SSL_CAMELLIA            (SSL_CAMELLIA128|SSL_CAMELLIA256)
+# define SSL_CHACHA20            (SSL_CHACHA20POLY1305)
```
Comment (ghedo) [inline ssl/ssl_locl.h]: Would it make sense to have a `SSL_POLY1305` alias as well?

Comment (kroeckx) [inline ssl/ssl_locl.h]: We don't have a "GCM" either, but we do have an "AESGCM".  I guess an "AEAD" could be useful as well.  But that can all be done in different patches.

**Comment by richsalz** [comment]
No, you go for it :)

File: ssl/ssl_locl.h (hunk 2 above)
Comment (richsalz) [inline ssl/ssl_locl.h]: I hope we don't add more aliases.

**Comment by kroeckx** [comment]
@richsalz: Can you please at least indicate that you've reviewed it, other than asking me to merge it?

**Comment by richsalz** [comment]
+1
